python/ribasim/ribasim/input_base.py

```diff
@@ -202,6 +216,44 @@ def node_ids(self) -> set[int]:
 
         return node_ids
 
+    def offset_node_ids(self, offset_node_id: int) -> "TableModel[TableT]":
```

Comment:
This does also create a copy, which could be reflected in the name. Maybe we want to have a mutable version as well?